include/oneapi/dpl/internal/common_config.h

```diff
@@ -16,6 +16,14 @@
 #ifndef _ONEDPL_COMMON_CONFIG_H
 #define _ONEDPL_COMMON_CONFIG_H
 
+#if !defined(__cplusplus)
+#    error "oneDPL requires correct state of __cplusplus : it's undefined"
+#endif
+
+#if __cplusplus == 199711L
+#    error "oneDPL requires correct state of __cplusplus : use /Zc:__cplusplus for Microsoft C++ compiler"
+#endif
+
 #if __cplusplus < 201703L
 #    error "oneDPL requires the C++ language version not less than C++17"
```

Comment:
MSVC compiler
We are here for command line like this:  -std:c++14  -Zc:__cplusplus